smtp/address.ts

```diff
@@ -5,7 +5,7 @@ interface AddressToken {
 
 export interface AddressObject {
 	address?: string;
-	name?: string;
+	name?: string | undefined;
```

Comment:
`name?: string` and `name?: string | undefined` are equivalent. can we do without this?

https://www.typescriptlang.org/play?#code/ATCWDsBcFMCcDMCGBjawAq0DOkDyAjAK2mUmAG8AoEEcRAW2gH4AuYHWCAc2uAF9KvZAHtwOYDBxtMOAsVLAAvBV60G0NgFdwAE2jwI0HbwGUgA

Reply:
it supports the [`exactOptionalPropertyTypes`](https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes) tsconfig option, which was added in 4.4: 

https://www.typescriptlang.org/play?exactOptionalPropertyTypes=true#code/ATCWDsBcFMCcDMCGBjawAq0DOkDyAjAK2mUmAG8AoEEcRAW2gH4AuYHWCAc2uAF9KvZAHtwOYDBxtMOAsVLAAvBV60G0NgFdwAE2jwI0HbwGUgA